homeassistant/components/wiz/config_flow.py

```diff
@@ -137,31 +138,36 @@ async def async_step_user(
         """Handle a flow initialized by the user."""
         errors = {}
         if user_input is not None:
-            if not (host := user_input[CONF_HOST]):
-                return await self.async_step_pick_device()
-            bulb = wizlight(host)
             try:
-                mac = await bulb.getMac()
-                bulbtype = await bulb.get_bulbtype()
-            except WizLightTimeOutError:
-                errors["base"] = "bulb_time_out"
-            except ConnectionRefusedError:
-                errors["base"] = "cannot_connect"
-            except WizLightConnectionError:
-                errors["base"] = "no_wiz_light"
-            except Exception:  # pylint: disable=broad-except
-                _LOGGER.exception("Unexpected exception")
-                errors["base"] = "unknown"
+                socket.inet_aton(user_input[CONF_HOST])
```

Comment:
This is doing I/O inside the event loop.

Reply:
Also, for IP validation, use `from ipaddress import ip_address`

Reply:
We also have a helper `from homeassistant.util.network import is_ip_address`

Reply:
I expected something like that. Let's do my homework 😄